api/tasks.go

```diff
@@ -1110,3 +1112,10 @@ func (t *TaskCSIPluginConfig) Canonicalize() {
 		t.HealthTimeout = 30 * time.Second
 	}
 }
+
+// WorkloadIdentity is the jobspec block which determines if and how a workload
+// identity is exposed to tasks.
+type WorkloadIdentity struct {
+	Env  *bool `hcl:"env,optional"`
+	File *bool `hcl:"file,optional"`
```

Comment:
Now that these default false, do they need to be pointers?

Reply:
They don't, but I kind of wanted to reserve the ability to distinguish between `false` and `unset` in the future (without a breaking API change).

For example if we add `file_name = "..."` or `file_owner = "..."` parameters, do we still need to make set `file` explicitly? We may even want to change the default to `true` in the future if the unauthenticated HTTP endpoints issue can be addressed.

I'm fine removing the pointer though if this is too strange.

Reply:
We've shipped so many nil pointer panics already I would lean toward using them only when they're necessary. Having these as pointers is also annoying to the API user, who now has to do the 

```
var b bool
t.Env = &b
```

hoop jumping.